### DISCUSSION TITLE
Connect a USB WiFi adapter

### DISCUSSION THREAD

**Discussion by Wini-Buh** [hathach/tinyusb#1175] [General]
I would like to connect an EDIMAX N150 Wi-Fi Nano USB Adapter to an embedded system (the USB adapter uses a Realtek RTL8192CU chip, I don't know if this is important).

What do I need to get it working (which device class uses this kind of adapter) ?
I guess I need also to run a TCP/IP stack (like lwIP) on my embedded system, is this correct ?

Any help would be welcome

**Comment by hathach** [comment]
- sound like a great project, which mcu you are using, I guess the RX72n ? 
- You need a host driver for your mcu
- depending on your device, you need an host class driver for it. I suspect it is rndis. Could you provide its device descriptor here, maybe some kernel log and/or driver info from windows/linux
- yeah, you will need an tcp/ip stack, the device only serves as transportation (rndis/ecm/ncm)

**Reply by Wini-Buh** [reply]
Thanks for your reply.
Apologies for not mentioning that this request is for a friend of mine who does not have a Github login.

No it's not for a Rx72N, it's for an STM32 controller.

BTW: I made some changes on the RNDIS sources to be endian independent (for the Rx72N for example) and will make a PR of it soon.

**Reply by hathach** [reply]
Unfortunately there is no host driver for stm32 as well, current only nxp with ehci/ohci driver along with rp2040 has host driver. The host stack does not receive as much developing time as device one.  Looking forward your endian fix, I rarely use BE mcu.
